test/Starcounter.Database.Extensions.IntegrationTests/NestedTransactorTests.cs

```diff
@@ -0,0 +1,36 @@
+using Microsoft.Extensions.DependencyInjection;
+using Xunit;
+
+namespace Starcounter.Database.Extensions.IntegrationTests
+{
+    public sealed class NestedTransactorTests : ServicedTests
+    {
+        [Fact]
+        public void AllowNestedTransaction()
+        {
+            var transactor = CreateServices(
+                s => s.Decorate<ITransactor, NestedTransactor>())
+                .GetRequiredService<ITransactor>();
+
+            transactor.Transact(db =>
+            {
+                transactor.Transact(db => { });
+            });
+        }
+
+        [Fact]
+        public void ShouldRenderContextThatIndicatesNesting()
```

Comment:
Please add a test for an async delegate that would fail without `AsyncLocal`

Reply:
Something like:

```cs
        [Fact]
        public void AllowNestingInAsync()
        {
            var transactor = CreateServices(
                s => s.Decorate<ITransactor, NestedTransactor>())
                .GetRequiredService<ITransactor>();

            transactor.Transact(async db =>
            {
                await System.Threading.Tasks.Task.Yield();
                transactor.Transact(db => Assert.True(db.IsNested()) );
            }).Wait();
        }
```

Reply:
Good suggestion. Done.